docs/index.asciidoc

```diff
@@ -0,0 +1,231 @@
+END - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+
+[id="plugins-{type}s-{plugin}"]
+
+=== Google Cloud Pub/Sub Output Plugin
+
+include::{include_path}/plugin_header.asciidoc[]
+
+==== Description
+
+A Logstash plugin to upload log events to https://cloud.google.com/pubsub/[Google Cloud Pubsub].
+Events are batched and uploaded in the background for the sake of efficiency.
+Message payloads are serialized JSON representations of the events.
+
+Example use-cases:
+
+ * Stream events to Dataproc via Pub/Sub for real-time analysis.
+ * Forward events from an on-prem datacenter to the Logstash in the cloud.
+ * Use Pub/Sub as an scalable buffer to even out event flow between processing steps.
+
+Note: While this project is partially maintained by Google, this is not an official Google product.
+
+=== Environment Configuration
+
+To use this plugin, you must create a
+https://developers.google.com/storage/docs/authentication#service_accounts[service account]
+and grant it the publish permission on a topic.
+You MAY also use the https://cloud.google.com/docs/authentication/production[Application Default Credentials]
+assigned to a compute instance.
+
+The Pub/Sub topic __must__ exist before you run the plugin.
+
+
+==== Example Configurations
+
+===== Basic
+
+A basic configuration which only includes a project, topic, and JSON key file:
+
+[source,ruby]
+------------------------------------------------------------------------------
+output {
+  google_pubsub {
+    # Required attributes
+    project_id => "my_project"
+    topic => "my_topic"
+
+    # Optional if you're using app default credentials
+    json_key_file => "service_account_key.json"
+  }
+}
+------------------------------------------------------------------------------
+
+
+===== High Volume
+
+If you find that uploads are going too slowly, you can increase the message batching:
+
+[source,ruby]
+------------------------------------------------------------------------------
+output {
+  google_pubsub {
+    project_id => "my_project"
+    topic => "my_topic"
+    json_key_file => "service_account_key.json"
+
+    # Options for configuring the upload
+    message_count_threshold => 10000
+    delay_threshold_secs => 10
+    request_byte_threshold => 5MB
+  }
+}
+------------------------------------------------------------------------------
+
+
+===== Attributes
+
+You can attach additional attributes to each request.
+For example, you could attach a datacenter label to a log message to help with debugging:
+
+[source,ruby]
+------------------------------------------------------------------------------
+output {
+  google_pubsub {
+    project_id => "my_project"
+    topic => "my_topic"
+    json_key_file => "service_account_key.json"
+
+
+    attributes => {"origin" => "pacific-datacenter"}
+  }
+}
+------------------------------------------------------------------------------
+
+
+====== Different Codecs
+
```

Comment:
This heading exceeds the levels we can render.  Can you remove one `=` to set it to `===== Different Codecs`?
If this heading must be one level down from Attributes, we can try surrounding heading  with `*` to make it bold.